dashboard/app/controllers/report_abuse_controller.rb

```diff
@@ -19,70 +19,82 @@ def to_honeybadger_context
 class ReportAbuseController < ApplicationController
   AGE_CUSTOM_FIELD_ID = 24_024_923
 
+  # Projects that are created by users with project validator permissions are
+  # blocked from abuse reports because they are created internally and we know
+  # they are safe. This reduces spamming of the report abuse feature.
+  def protected_project?
+    return false if params[:channel_id].blank?
+    owner, _storage_app_id = storage_decrypt_channel_id(params[:channel_id])
+    project_owner = User.find(owner)
+    project_owner.permission?(UserPermission::PROJECT_VALIDATOR)
+  end
+
   def report_abuse
-    unless Rails.env.development? || Rails.env.test?
-      subject = FeaturedProject.featured_channel_id?(params[:channel_id]) ?
-        'Featured Project: Abuse Reported' :
-        'Abuse Reported'
-      response = HTTParty.post(
-        'https://codeorg.zendesk.com/api/v2/tickets.json',
-        headers: {"Content-Type" => "application/json", "Accept" => "application/json"},
-        body: {
-          ticket: {
-            requester: {
-              name: (params[:name] == '' ? params[:email] : params[:name]),
-              email: params[:email]
-            },
-            subject: subject,
-            comment: {
-              body: [
-                "URL: #{params[:abuse_url]}",
-                "abuse type: #{params[:abuse_type]}",
-                "user detail:",
-                params[:abuse_detail]
-              ].join("\n")
-            },
-            custom_fields: [{id: AGE_CUSTOM_FIELD_ID, value: params[:age]}],
-            tags: (params[:abuse_type] == 'infringement' ? ['report_abuse', 'infringement'] : ['report_abuse'])
-          }
-        }.to_json,
-        basic_auth: {username: 'dev@code.org/token', password: Dashboard::Application.config.zendesk_dev_token}
-      )
-      raise ZendeskError.new(response.code, response.body) unless response.success?
-    end
+    unless protected_project?
+      unless Rails.env.development? || Rails.env.test?
```

Comment:
why is this needed? we try to keep these kinds of checks out of our application code because it makes them less testable. if you do need to perform this check, please consider using a CDO variable which can then be stubbed during testing.

Reply:
Do you mean why is `unless Rails.env.development? || Rails.env.test?`needed? It's there so we don't spam Zendesk with tickets from test and development.  I don't think I understand what you're suggesting.

Reply:
sorry, I missed that this check was in there already. I guess it is fine to leave, then.

I was suggesting defining a variable like `CDO.send_abuse_reports` which would be true in production and false in development / test.

Reply:
Oh, ok.  Got it. Thanks for clarifying. That makes sense, now.